### PR TITLE
Submit/stagit gopher

### PR DESCRIPTION
**Pull request by hcssmith**
stagit-gopher init at version 0.9.3

###### Motivation for this change
So I can use stagit-gopher on Nixos

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/stagit-gopher/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, libgit2, fetchgit }:
+
+stdenv.mkDerivation rec {
+                    name = "stagit-gopher";
```
Comment (IvarWithoutBones) [inline pkgs/development/tools/stagit-gopher/default.nix:4]: You should be using `pname` here.

File: pkgs/development/tools/stagit-gopher/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                    src = fetchgit {
+                        url = "git://git.codemadness.org/stagit-gopher";
+                        rev = version;
+                        sha256 = "0dcyi3k900jb8p5qhgfk91gvplblysgfz1vyn6pqrlk931hpac80";
+                    };
+
+                    makeFlags = [ "PREFIX=$(out)" ];
+
+                    buildInputs = [ libgit2 ];
+
```
Comment (IvarWithoutBones) [inline pkgs/development/tools/stagit-gopher/default.nix:16]: Second newline isn't needed

**Review by IvarWithoutBones — COMMENTED** [review]
Few minor nits, otherwise LGTM

Indentation is off, should be two spaces i think.

Commits don't follow the [contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) (should be squashed, with the message as described in there). Should note that the maintainer list addition should be in a seperate commit as well.

**Comment by hcssmith** [comment]
OK @IvarWithoutBones , So I should fix the formatting in the actual code files, squash the commits and do a separate pull request for the maintainers file?

**Comment by IvarWithoutBones** [comment]
Yep, except that the maintainer file changes can be in this PR, it should just be in a different commit.

**Comment by hcssmith** [comment]
@IvarWithoutBones, Right think I've got it this time. Maintainer and package in in separate commits and I think I've got the formatting sorted this time. Sorry about the slight ineptitude with getting the commits correctly squashed together. Must have taken about three or four resets

**Review by IvarWithoutBones — APPROVED** [review]
> Sorry about the slight ineptitude with getting the commits correctly squashed together. Must have taken about three or four resets

No worries, git can be a bit confusing sometimes

LGMT :)

Output from `nixpkgs-review`:
```
1 package built:
stagit-gopher
```

File: pkgs/development/tools/stagit-gopher/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+stdenv.mkDerivation rec {
+  pname = "stagit-gopher";
+  version = "0.9.3";
+
+  src = fetchgit {
+    url = "git://git.codemadness.org/stagit-gopher";
+    rev = version;
+    sha256 = "0dcyi3k900jb8p5qhgfk91gvplblysgfz1vyn6pqrlk931hpac80";
+    };
```
Comment (IvarWithoutBones) [inline pkgs/development/tools/stagit-gopher/default.nix:11]: Indentation seems to be wrong here

File: pkgs/development/tools/stagit-gopher/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  buildInputs = [ libgit2 ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://git.codemadness.org/stagit-gopher/file/README.html";
+    description = "Static git site generator for gopher";
+    license = licenses.mit;
+    platforms = platforms.all;
+    maintainers = with maintaininers; [ hcssmith ];
```
Comment (IvarWithoutBones) [inline pkgs/development/tools/stagit-gopher/default.nix:22]: Typo here, causing ofborg to fail.
```suggestion
    maintainers = with maintainers; [ hcssmith ];
```

**Review by IvarWithoutBones — COMMENTED** [review]
Apologies, looks like i missed a few more nits

File: pkgs/development/tools/stagit-gopher/default.nix (hunk 4 above)
Comment (hcssmith) [reply pkgs/development/tools/stagit-gopher/default.nix]: OK, I've corrected the typo, not sure how i missed that given that I wrote it two words before, and I have moved the } back two spaces which I think is what you meant.

**Review by IvarWithoutBones — APPROVED** [review]

**Pull request closed without merging**